controllers/employeeController.js

```
@@ -1,18 +1,89 @@
+    const passwordErrors = validatePassword(password);
+    if (passwordErrors.length > 0) {
+      console.log("password not valid");
+      return res.status(400).json({ error: 'Invalid password', passwordErrors });
+    }
+
+    // Check if the email is already registered
+    const existingUser = await Employee.findOne({ email });
+    if (existingUser) {
+      return res.status(409).json({ success: false, message: 'Email already registered' });
+    }
+
+    // Encrypt the password
+    const hashedPassword = await bcrypt.hash(password, 10);
+    // Create a new employee with the encrypted password
+    const employee = await Employee.create({
+      name,
+      email,
+      password : hashedPassword,
+      age,
+      gender,
+      address,
+      designation,
+      mobile
+    });
+
+    res.status(201).json({ success: true, message: 'User created successfully' });
+
+  } catch (error) {
+    res.status(500).json({ success: false, error: error.message });
+  }
+}
+
+// Log in a user and generate a JWT token
+async function login(req, res) {
+  try {
+    const { email, password } = req.body;
+
+    // Find the employee by email
+    const employee = await Employee.findOne({ email });
+    if (!employee) {
+      return res.status(401).json({ success: false, message: 'Invalid email or password' });
+    }
+
+    // Validate the password
+    const isPasswordValid = await bcrypt.compare(password, employee.password);
+    if (!isPasswordValid) {
+      return res.status(401).json({ success: false, message: 'Invalid email or password' });
+    }
+
+    // Generate a JWT token
+    const token = jwt.sign({ id: employee._id }, process.env.ACCESS_TOKEN, { expiresIn: '1h' });
+
+    res.json({ success: true, token });
+  } catch (error) {
+    res.status(500).json({ success: false, error: error.message });
+  }
+}
+
 // Add employee details to the database
 async function addEmployee(req, res) {
   try {
     const employee = await Employee.create(req.body);
```

When adding an employee, please check to see if the person already exists; if so, please return an error.